ConsoleApp/ConsoleApp/Program.cs

```diff
@@ -0,0 +1,217 @@
+                    phoneBook[i - 1] = (splitted[0], splitted[1], splitted[2]);
+                }
+
+                return phoneBook;
+            }
+
+            static void SearchByName()
+            {
+                Console.Clear();
+                Console.WriteLine("Search condition - if first/last name contains search term you will see it");
+
+                Console.WriteLine("Enter Search Term...");
+                var searchTerm = Console.ReadLine();
+                bool found = false;
+
+                foreach (var (firstName, lastName, phoneNumber) in ReadPhoneBook())
+                {
+                    if (firstName.Contains(searchTerm, StringComparison.OrdinalIgnoreCase) ||
+                        lastName.Contains(searchTerm, StringComparison.OrdinalIgnoreCase))
+                    {
+                        Console.WriteLine($"Found user {firstName} {lastName} with phone {phoneNumber}");
+                        found = true;
+                    }
+                }
+
+                if (!found)
+                {
+                    Console.WriteLine("Not such users");
+                }
+
+                Wait();
+            }
+
+            //Search by Phone Number
+            //Similar to SearchByName() 
+            static void SearchByPhoneNumber()
+            {
+                Console.Clear();
+                Console.WriteLine("Search condition - if phone number contains search digits we will find phone number ");
+
+                Console.WriteLine("Enter numbers...");
+                var searchTerm = Console.ReadLine();
+                bool found = false;
+
+                foreach (var (firstName, lastName, phoneNumber) in ReadPhoneBook())
+                {
+                    if (phoneNumber.Contains(searchTerm))
+                    {
+                        Console.WriteLine($"Found user {firstName} {lastName} with phone {phoneNumber}");
+                        found = true;
+                    }
+                }
+
+                if (!found)
+                {
+                    Console.WriteLine("Not such users");
+                }
+
+                Wait();
+            }
+
+            static void UpdatePhoneNumber()
+            {
+                Console.Clear();
+
+                Console.WriteLine("Write the FirstName and LastName where we will change the phone ");
+                string lookFirstName = Console.ReadLine();
+                string lookLastName = Console.ReadLine();
+
+                bool found = false;
+                (string firstName, string lastName, string phoneNumber)[] phoneBook = ReadPhoneBook();
+
+                //Here we looking our contact and as soon as we found, we change it and write in fle
+                for (int i = 0; i < phoneBook.Length; i++)
+                {
+                    if (phoneBook[i].firstName.ToLower() == lookFirstName.ToLower() && phoneBook[i].lastName.ToLower() == lookLastName.ToLower())
```

Comment:
instead of it better use string.Equals(str1, str2, StringComparison.OrdinalIgnoreCase)